### ISSUE TITLE
compiler warnings in new cljs versions

### ISSUE THREAD

**Issue by thedavidmeister** [noprompt/garden#144]
```
       Aug 22, 2017 9:57:09 AM com.google.javascript.jscomp.LoggerErrorManager println
       WARNING: /app/.boot/cache/tmp/tmp/build_249b3a126e0f26fe96d0a5c79f919ce5/thedavidmeister-estimate-work-1535383fd184345db3dec0c34380e4682d8e6db5/fg/-b1ra1p/index.0f76fa787e214660aceb08719d8e4c33.html.out/garden/types.js:306: WARNING - Keywords and reserved words are not allowed as unquoted property names in older versions of JavaScript. If you are targeting newer versions of JavaScript, set the appropriate language_in option.
       return (!((other15033 == null))) && ((this15032__$1.constructor === other15033.constructor)) && (cljs.core._EQ_.cljs$core$IFn$_invoke$arity$2(this15032__$1.function,other15033.function)) && (cljs.core._EQ_.cljs$core$IFn$_invoke$arity$2(this15032__$1.args,other15033.args)) && (cljs.core._EQ_.cljs$core$IFn$_invoke$arity$2(this15032__$1.__extmap,other15033.__extmap));
```

**Comment by rap1ds** [comment]
@thedavidmeister I saw the same warning today when updating ClojureScript from 1.9.292 to 1.9.908.

To fix the issue, set `language-in` and `language-out` compiler options to `:ecmascript5`.

```
;; project.clj

:cljsbuild
  {:builds 
    [{:id "dev"
      ;; ...
      :compiler {
                 ;; ...

                 :language-in     :ecmascript5
                 :language-out    :ecmascript5

                 ;; ...
                 }}
```

The default for `:language-in` and `:language-out` is `:ecmascript3`. Changing it to `:ecmascript5` fixes the issue, but obviously that's not an option if you really need to target very old browsers, like IE8.

**Comment by noprompt** [comment]
@rap1ds @thedavidmeister Is this specifically a Garden issue? and, if so, can either of you provide a patch? @rap1ds it sounds like you might have some knowledge in this area. I'm very busy with work and family these days and I simply don't have the time to invest in researching this issue. Thanks!

**Comment by thedavidmeister** [comment]
@noprompt garden is the only thing throwing this warning for me.

i don't exactly know, but i'm _assuming_ it's the use of the name `function` (a reserved word in JS) here https://github.com/noprompt/garden/blob/master/src/garden/types.cljc#L6

**Comment by noprompt** [comment]
@thedavidmeister ugh, there's an old Jira ticket out there for this problem. It would appear the solution, for now, would be to rename that parameter to something else (`funxion` might be fun) and make sure that's handled anywhere else in the library.

**Comment by rap1ds** [comment]
Would you consider the `CSSFunction` defrecord as a part of the Garden "public API"? I mean, if the name `function` is changed, is that then a breaking change? At least I'm using code like `(:magnitude (px 1234))` in my app to access the pixel values (or am I doing it wrong? I didn't find any other way to access the value). So if someone is using the `function` the same way, the chande of the name would break the code.

However, changing the `function` name sounds like a good idea to me, even if it's a breaking change. In my own app I'm happy with changing the language settings to ecmascript5, but since ecmascript3 is the CLJS compiler default, it makes sense to fix the library so that it works without warnings with the default compiler options.

**Comment by noprompt** [comment]
@rap1ds Apologies for my delayed response. I think it's admissible to change the name here. It's a breaking change for sure. To the extent it happens to break someone upon upgrade, they can share their grievances here.

Feel free to submit a patch in response to this. I'm busy at the moment and can't get to it now.

**Comment by danielcompton** [comment]
I've opened a PR to fix this in #158.
